### ISSUE TITLE
Cannot execute `dark_evaluation`

### ISSUE THREAD

**Issue by dneise** [calispac/digicampipe#75]
As mentioned in #74 I think users of scripts should not have to modify them to get them to work, but to call them with command line arguments.

Nevertheless, I tried to execute `dark_evaluation.py` on the crab files I downloaded using Etiennes script.

This is the diff of `dark_evaluation.py`
```diff
diff --git a/dark_evaluation.py b/dark_evaluation.py
index fe2643a..d242d0e 100644
--- a/dark_evaluation.py
+++ b/dark_evaluation.py
@@ -7,14 +7,24 @@ from digicampipe.utils import geometry
 import matplotlib.pyplot as plt
 import numpy as np
 
+import pkg_resources
+import os
 
 if __name__ == '__main__':
     # Data configuration
 
-    directory = '/home/alispach/data/CRAB_01/'
-    filename = directory + 'CRAB_01_0_000.%03d.fits.fz'
+    directory = '/home/dneise/ctasoft/sst1m_crab/'
+    filename = directory + 'SST1M01_20171030.%03d.fits.fz'
     file_list = [filename % number for number in [2]]
-    camera_config_file = '/home/alispach/ctasoft/CTS/config/camera_config.cfg'
+
+    camera_config_file = pkg_resources.resource_filename(
+        'digicampipe',
+        os.path.join(
+            'tests',
+            'resources',
+            'camera_config.cfg'
+        )
+    )
     dark_filename = 'dark.npz'
 
     trigger_filename = 'bias_curve_dark.npz'
@@ -27,14 +37,13 @@ if __name__ == '__main__':
     digicam_geometry = geometry.generate_geometry_from_camera(camera=digicam)
 
     # Define the event stream
-    data_stream = event_stream(file_list=file_list, expert_mode=True, camera_geometry=digicam_geometry)
+    data_stream = event_stream(file_list=file_list, expert_mode=True, camera_geometry=digicam_geometry, camera=digicam)
     data_stream = filter.set_patches_to_zero(data_stream, unwanted_patch=unwanted_patch)
     # Fill the flags (to be replaced by Digicam)
     data_stream = filter.filter_event_types(data_stream, flags=[8])
 
     data_stream = save_bias_curve(data_stream,
                                   thresholds=thresholds,
-                                  camera=digicam,
                                   blinding=blinding,
                                   output_filename=directory + trigger_filename,
                                   unwanted_cluster=unwanted_cluster)
```

... wait am I probably working with an old version here or so? Let me reinstall everything quickly.

**Comment by dneise** [comment]
Okay I freshly installed a new copy of digicampipe. 

And I still get the same error:
```
(digicampipe) dneise@lair:~/ctasoft/digicampipe$ ipython dark_evaluation.py 
Open has failed. Error: Table RunHeader could not be found in input file. Aborting. while reading file /home/dneise/ctasoft/sst1m_crab/SST1M01_20171030.002.fits.fz and table RunHeader
/home/dneise/ctasoft/digicampipe/digicampipe/io/protozfitsreader.py:185: UserWarning: /home/dneise/ctasoft/sst1m_crab/SST1M01_20171030.002.fits.fz has no RunHeader
  "{} has no RunHeader".format(self.fname))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/ctasoft/digicampipe/dark_evaluation.py in <module>()
     41     data_stream = save_dark(data_stream, directory + dark_filename)
     42 
---> 43     for i, data in enumerate(data_stream):
     44 
     45         print(i)

~/ctasoft/digicampipe/digicampipe/io/save_adc.py in save_dark(data_stream, output_filename, n_events)
    105         iter_events = range(n_events)
    106 
--> 107     for event, i in zip(data_stream, iter_events):
    108 
    109         for telescope_id in event.r0.tels_with_data:

~/ctasoft/digicampipe/digicampipe/io/save_bias_curve.py in save_bias_curve(data_stream, thresholds, output_filename, n_events, blinding, by_cluster, unwanted_cluster)
     24             if i == 0:
     25 
---> 26                 n_clusters = trigger_input_7.shape[0]
     27                 cluster_rate = np.zeros((n_clusters, thresholds.shape[-1]))
     28 

TypeError: 'getset_descriptor' object is not subscriptable
```

**Comment by dneise** [comment]
Okay so the problem seems to be, that `R0CameraContainer.trigger_input_7` is never being filled with any value by the `zfits_event_source`. 

I feared that I somehow messed this up in one of my PRs in the past, so I looked at bit at historic versions of the `zfits_event_source` in the git history, but so far I was not able to find a version which assigns a value to `trigger_input_7` somewhere here:

https://github.com/calispac/digicampipe/blob/39ab15dcd16179489ecf172db17ce73d58dce768/digicampipe/io/zfits.py#L64-L89

**Comment by calispac** [comment]
In digicampipe/io/save_bias_curve.py `trigger_input_7` should be `trigger_input_traces`. It represents the "patch view" of the camera (sum of 3 neighbor pixels basically).

**Comment by dneise** [comment]
Okay, then I have a question. 

How should this look like?
https://github.com/calispac/digicampipe/blob/39ab15dcd16179489ecf172db17ce73d58dce768/digicampipe/io/containers.py#L121-L126

should trigger_input_7 and _19 be deleted?

---- 
Actually I have two questions.

Why am I running into this issue today and not anybody else who is working with digicampipe much more than me? Am i doing something fundamentally wrong?

**Comment by calispac** [comment]
> In digicampipe/io/save_bias_curve.py `trigger_input_7` should be `trigger_input_traces`. It represents the "patch view" of the camera (sum of 3 neighbor pixels basically).

Sorry I was wrong....
`trigger_input_traces` are the patch traces provided by digicam. (sum of 3 neighboring pixels)
`trigger_input_7` are the traces that are directly compared to the threshold to make the trigger decision (for the trigger algorithm taking a group of 7 neighboring patches).

**Comment by Etienne12345** [comment]
Then shoudln't "trigger_input_7" rather be called "trigger_output_7" ?? I mean, it is the output of the digicam computation, right ?

**Comment by dneise** [comment]
> Then shoudln't "trigger_input_7" rather be called "trigger_output_7" ?? I mean, it is the output of the digicam computation, right ?

I also do not yet get the wording here ...

**Comment by calispac** [comment]
the computation of the `trigger_input_7` should be done before running `save_bias_curve` in `dark_evaluation.py` with this function: 

https://github.com/calispac/digicampipe/blob/39ab15dcd16179489ecf172db17ce73d58dce768/digicampipe/calib/camera/r0.py#L17-L27

It seems that we deleted this line at some point or forgot to write it while doing some refactoring.

**Comment by calispac** [comment]
I agree wording is quite confusing here...

Just to list what we have concerning trigger: 

1. From the raw data (computed by DigiCam)
     - patch traces (sum of 3 neighboring pixel traces) : shape = (n_pixels/3, n_{trace samples}) = (432, 50), dtype=uint8
     - trigger output from trigger algorithm 7 : shape = (n_pixels/3, n_trace samples}) = (432, 50), dtype=bool
    - trigger output from trigger algorithm 19 : shape = (n_pixels/3, n_{trace samples}) = (432, 50), dtype=bool

2. What is computed by DigiCam but not in the raw data

    - cluster_7 traces (sum of 7 neighboring patch) : shape = (n_pixels/3, n_{trace samples}) = (432, 50), dtype=uiny8

    - cluster_19 traces (sum of 19 neighboring patch) : shape = (n_pixels/3, n_{trace samples}) = (432, 50), dtype=uint8

Generally `input` should correspond to what enters the ">" comparison and "output" the result of that comparison. Naming suggestions are welcome

**Comment by calispac** [comment]
@dneise Do you want me to make a PR to fix this in `dark_evaluation.py` ?

**Comment by dneise** [comment]
I can do it .. and you review .. okay?

**Comment by dneise** [comment]
I think this is fixed by #77 once merged. 
Some details about command line interface are still to be decided, but generally #77 fixes this issue.
